core/pkg/ingress/types.go

```diff
@@ -0,0 +1,288 @@
+	Repository string `json:"repository"`
+}
+
+// Configuration holds the definition of all the parts required to describe
+type Configuration struct {
+	// HealthzURL location of the URI that is used to know health state
+	// of the ingress controller
+	HealthzURL string `json:"healthzUrl"`
+	// Backends
+	Backends []*Backend `json:"namespace"`
+	// Servers
+	Servers []*Server `json:"servers"`
+	// TCPEndpoints
+	// +optional
+	TCPEndpoints []*Location `json:"tcpEndpoints,omitempty"`
+	// UPDEndpoints
+	// +optional
+	UPDEndpoints []*Location `json:"udpEndpoints,omitempty"`
+	// PassthroughBackend
+	// +optional
+	PassthroughBackend []*SSLPassthroughBackend `json:"passthroughBackend,omitempty"`
+}
+
+// Server describes a virtual server.
+type Server struct {
+	// Hostname returns the FQDN of the server
+	Hostname string `json:"hostname"`
+	// SSL indicates if the virtual server is configured for TLS
+	SSL bool `json:"ssl"`
+	// SSLPassthrough indicates if the TLS termination is realized in
+	// the server or in the remote endpoint
+	SSLPassthrough bool `json:"sslPassthrough"`
+	// SSLCertificate path to the SSL certificate on disk
+	SSLCertificate string `json:"sslCertificate"`
+	// SSLPemChecksum returns the checksum of the certificate file on disk.
+	// There is no restriction in
+	// This helps to determine if the secret changed without the use of
+	// file system notifications
+	SSLPemChecksum string `json:"sslPemChecksum"`
+	// Locations list of URIs configured in the server
+	Locations []*Location `json:"locations,omitempty"`
+}
+
+// Location describes an URI inside a server.
+// Also contains additional information about defined annotations in
+// the Ingress rule.
+//
+// Important:
+// The implementation of annotations is optional
+type Location struct {
+	// Path is an extended POSIX regex as defined by IEEE Std 1003.1,
+	// (i.e this follows the egrep/unix syntax, not the perl syntax)
+	// matched against the path of an incoming request. Currently it can
+	// contain characters disallowed from the conventional "path"
+	// part of a URL as defined by RFC 3986. Paths must begin with
+	// a '/'. If unspecified, the path defaults to a catch all sending
+	// traffic to the backend.
+	Path string `json:"path"`
+	// IsDefBackend indicates if service specified in the Ingress
+	// contains active endpoints or not.
+	IsDefBackend bool `json:"isDefBackend"`
+	// Backend describes the endpoints to use
+	Backend Backend `json:"backend"`
+	// BasicDigestAuth returns authentication configuration for
+	// an Ingress rule
+	// +optional
+	BasicDigestAuth auth.BasicDigest `json:"basicDigestAuth,omitempty"`
+	// EnableCORS indicates if path must support CORS
+	// +optional
+	EnableCORS bool `json:"enableCors,omitempty"`
+	// ExternalAuth indicates the access to this location requires
+	// authentication using an external provider
+	// +optional
+	ExternalAuth authreq.External `json:"externalAuth,omitempty"`
+	// RateLimit describes
```

Comment:
please finish the sentence, I guess it describes a rate limit for this specific path?